### PR TITLE
[TEST] PromQL: Adds tests with histograms for simple functions

### PR DESCRIPTION
**Pull request by NeerajGartia21**
Part of #13934.

This PR adds tests with histograms and verifies that simple functions like `exp`, `ln`, `log2`, `log10`, `sqrt` and `sgn` ignores histograms.

CC: @beorn7

### CODE REVIEW

File: promql/promqltest/testdata/functions.test [hunk 1]
```diff
@@ -494,6 +494,7 @@ load 5m
 	test_sgn{src="sgn-d"}	-50
 	test_sgn{src="sgn-e"}	0
 	test_sgn{src="sgn-f"}	100
+    test_sgn{src="sgn-histogram"} {{schema:1 sum:1 count:1}}
```
Comment (beorn7) [inline promql/promqltest/testdata/functions.test:497]: This line uses spaces, but the previous lines use tabs. Depending on the tab width, that might lead to weird alignment. (E.g. GH displays this file with 8-space tabs, so the lines above are too far indented.)

I recommend to use a tab in this line too. (Or change all lines to use spaces.)

File: promql/promqltest/testdata/functions.test [hunk 2]
```diff
@@ -1239,11 +1240,16 @@ clear
 load 5m
 	exp_root_log{l="x"} 10
 	exp_root_log{l="y"} 20
+    exp_root_log_h{l="z"} {{schema:1 sum:1 count:1}}
```
Comment (beorn7) [inline promql/promqltest/testdata/functions.test:1243]: This line uses spaces, but the previous lines use tabs. Depending on the tab width, that might lead to weird alignment. (E.g. GH displays this file with 8-space tabs, so the lines above are too far indented.)

I recommend to use a tab in this line too. (Or change all lines to use spaces.)

**Review by beorn7 — CHANGES_REQUESTED** [review]
Thank you very much. Just a tiny indentation issue.

**Comment by NeerajGartia21** [comment]
Fixed the indentaiton 🚀

File: promql/promqltest/testdata/functions.test (hunk 1 above)
Comment (NeerajGartia21) [reply promql/promqltest/testdata/functions.test]: Thanks. I'll fix this.

**Review by beorn7 — APPROVED** [review]

**Pull request merged**